### DISCUSSION TITLE
ERR no matching outgoing channel available for node

### DISCUSSION THREAD

**Discussion by HotCrypt** [lightningnetwork/lnd#8169] [General]
Hi, I'm trying to balance 2 channels (triangle ln.plus) but it doesn't work, here's what I tried:

`lncli addinvoice --amt 10000`

`lncli buildroute --amt 10000 --hops 0240b2b1f91d742327ebd34ea1129985ec98f9c08f7ab9a884a77ff3472897242e,038fdaa84188212e7aec321d443c3206a3ea4e009aa41f37be194154b1be6726c1,0368083d4e4705f261e365cb29bb65299e7c62fe3953b3f4820efce5af89b737be | jq -r '(.route.hops[-1] | .mpp_record) |= {payment_addr:"16bebb92e90d7e7bb3449ec5017e40404d26879e526b6df8471ab43376c076dc", total_amt_msat: "10000000"}' | lncli sendtoroute --payment_hash=b76ebfba2ce01dd4870b79c8f952c5c01bd8e714b514dc06f1ac09cd32cb06ed -
`
> [lncli] rpc error: code = Unknown desc = no matching outgoing channel available for node 0 (0368083d4e4705f261e365cb29bb65299e7c62fe3953b3f4820efce5af89b737be)
> [lncli] queryroutes output is empty

Logs:
> [WRN] CRTR: ShortChannelID=816133:31:1: cannot add outgoing htlc: commitment transaction dips peer below chan reserve: negative local balance 
> [ERR] CRTR: Cannot find policy with amt=10000231 mSAT for node 0368083d4e4705f261e365cb29bb65299e7c62fe3953b3f4820efce5af89b737be
> [ERR] RPCS: [/routerrpc.Router/BuildRoute]: no matching outgoing channel available for node 0 (0368083d4e4705f261e365cb29bb65299e7c62fe3953b3f4820efce5af89b737be)

Earlier today, but also failed:

`lncli buildroute --amt 10000 --hops 0240b2b1f91d742327ebd34ea1129985ec98f9c08f7ab9a884a77ff3472897242e,038fdaa84188212e7aec321d443c3206a3ea4e009aa41f37be194154b1be6726c1,0368083d4e4705f261e365cb29bb65299e7c62fe3953b3f4820efce5af89b737be | jq -r '(.route.hops[-1] | .mpp_record) |= {payment_addr:"42b4425c11166f45ab4b7522644d7ffe6c9ca6ed97aff302745a1716f8382cae", total_amt_msat: "10000000"}' | lncli sendtoroute --payment_hash=23144e2c4b1ef792e7d84d726b06123f43ba8fd3b1aa94654dd788c2e73560c2 -
`

> {
>     "attempt_id": "2018",
>     "status": "FAILED",
>     "route": {
>         "total_time_lock": 816345,
>         "total_fees": "0",
>         "total_amt": "10000",
>         "hops": [
>             {
>                 "chan_id": "897347723313741825",
>                 "chan_capacity": "20000",
>                 "amt_to_forward": "10000",
>                 "fee": "0",
>                 "expiry": 816305,
>                 "amt_to_forward_msat": "10000051",
>                 "fee_msat": "200",
>                 "pub_key": "0240b2b1f91d742327ebd34ea1129985ec98f9c08f7ab9a884a77ff3472897242e",
>                 "tlv_payload": true,
>                 "mpp_record": null,
>                 "amp_record": null,
>                 "custom_records": {},
>                 "metadata": ""
>             },
>             {
>                 "chan_id": "897226777042026496",
>                 "chan_capacity": "20000",
>                 "amt_to_forward": "10000",
>                 "fee": "0",
>                 "expiry": 816225,
>                 "amt_to_forward_msat": "10000000",
>                 "fee_msat": "51",
>                 "pub_key": "038fdaa84188212e7aec321d443c3206a3ea4e009aa41f37be194154b1be6726c1",
>                 "tlv_payload": true,
>                 "mpp_record": null,
>                 "amp_record": null,
>                 "custom_records": {},
>                 "metadata": ""
>             },
>             {
>                 "chan_id": "897242170196230145",
>                 "chan_capacity": "25575",
>                 "amt_to_forward": "10000",
>                 "fee": "0",
>                 "expiry": 816225,
>                 "amt_to_forward_msat": "10000000",
>                 "fee_msat": "0",
>                 "pub_key": "0368083d4e4705f261e365cb29bb65299e7c62fe3953b3f4820efce5af89b737be",
>                 "tlv_payload": true,
>                 "mpp_record": {
>                     "payment_addr": "42b4425c11166f45ab4b7522644d7ffe6c9ca6ed97aff302745a1716f8382cae",
>                     "total_amt_msat": "10000000"
>                 },
>                 "amp_record": null,
>                 "custom_records": {},
>                 "metadata": ""
>             }
>         ],
>         "total_fees_msat": "251",
>         "total_amt_msat": "10000251"
>     },
>     "attempt_time_ns": "1699622222504970161",
>     "resolve_time_ns": "1699622230787525305",
>     "failure": {
>         "code": "TEMPORARY_CHANNEL_FAILURE",
>         "channel_update": {
>             "signature": "298d6f36cc06e46324def4ba9c2e6ddcd68eae1ee21b5adba174ca50a1f75ce566fdb388d12d32e7f42d55136d3aadc8d1b893b5e94aedc81796fe53ccb92444",
>             "chain_hash": "6fe28c0ab6f1b372c1a6a246ae63f74f931e8365e15a089c68d6190000000000",
>             "chan_id": "897226777042026496",
>             "timestamp": 1699574402,
>             "message_flags": 1,
>             "channel_flags": 0,
>             "time_lock_delta": 40,
>             "htlc_minimum_msat": "1000",
>             "base_fee": 0,
>             "fee_rate": 20,
>             "htlc_maximum_msat": "17681400",
>             "extra_opaque_data": ""
>         },
>         "htlc_msat": "0",
>         "onion_sha_256": "",
>         "cltv_expiry": 0,
>         "flags": 0,
>         "failure_source_index": 1,
>         "height": 0
>     },
>     "preimage": ""
> }

I am well connected to these two nodes and the channels are active.
I did a sat send test with --keysend or --amp , it works fine.
I don't know why but since I use a hybrid tor/clearnet node I have a little more difficulty routing payments it seems, before with clearnet only, the routings were almost instantaneous, is this normal or did I misconfigured something?
I would like to point out that Tor works well, no errors.
Thanx.

**Comment by HotCrypt** [comment]
I wonder if its htlc is not set too low? My node is set to fee: 0/0 - htlc: 1/max

> [ERR] CRTR: Cannot find policy with amt=10000231 mSAT for node 0368083d4e4705f261e365cb29bb65299e7c62fe3953b3f4820efce5af89b737be


`lncli getchaninfo 897347723313741825`

> {
>     "channel_id": "897347723313741825",
>     "chan_point": "9edf4d02ffe136d9cc046cc4c8e4ec20eaf22e2352b0220c647cdc087bfa4162:1",
>     "last_update": 1699821231,
>     "node1_pub": "0240b2b1f91d742327ebd34ea1129985ec98f9c08f7ab9a884a77ff3472897242e",
>     "node2_pub": "0368083d4e4705f261e365cb29bb65299e7c62fe3953b3f4820efce5af89b737be",
>     "capacity": "20000",
>     "node1_policy": {
>         "time_lock_delta": 40,
>         "min_htlc": "1000",
>         "fee_base_msat": "0",
>         "fee_rate_milli_msat": "29",
>         "disabled": false,
>         "max_htlc_msat": "5893800",
>         "last_update": 1699812002,
>         "custom_records": {}
>     },
>     "node2_policy": {
>         "time_lock_delta": 40,
>         "min_htlc": "1",
>         "fee_base_msat": "0",
>         "fee_rate_milli_msat": "0",
>         "disabled": false,
>         "max_htlc_msat": "19800000",
>         "last_update": 1699821231,
>         "custom_records": {}
>     },
>     "custom_records": {}
> }
> 

Merci :)

**Reply by bitromortac** [reply]
Could you try to check what happens if you take an amount of 5000 sat and then 6000 sat? Your peer has indeed set a low max htlc size (if I interpret correctly that your node is `0368083d4e4705f261e365cb29bb65299e7c62fe3953b3f4820efce5af89b737be`).

**Reply by HotCrypt** [reply]
Hi 👋, 
-Yes, it's my node (Bitshirt).

I am in discussion by message with the other node.

-I just tried with 5000 and 6000 but still the same.

I sent a message to the other node to ask him to increase his htlc_max.
